### PR TITLE
Migrate CI from TeamCity to GitHub Actions

### PR DESCRIPTION
**Pull request by samanthagottlieb**
## What does this change?

This migrates CI from TeamCity to GitHub Actions.

Previously the deprecated `node-riffraff-artifact` module was used to compress the bundled file into a .zip file and upload files to Riff-Raff. This PR moves to using the Action `actions-riffraff` instead and does the zipping manually, as this Action [does not support compression](https://github.com/guardian/actions-riff-raff/issues/1).

Trello card is [here](https://trello.com/c/Yxm0Kdge/1448-migrate-our-projects-from-teamcity-to-github-actions).

## Further work
Node version 12 is defined in the `.nvmrc`, and version 14 is defined in cloudformation as the lambda runtime environment. Node 14 will be [deprecated](https://docs.aws.amazon.com/lambda/latest/dg/lambda-runtimes.html) for AWS lambdas on Nov 27, 2023, so this will need to be upgraded, as well as upgrading the application node version.

## How to test

This has been tested by comparing the zip files from a [previous successful build](https://riffraff.gutools.co.uk/deployment/request/deployFiles?projectName=cms-fronts%3A%3Afront-press-monitor&id=17) to a [new build](https://riffraff.gutools.co.uk/deployment/request/deployFiles?projectName=cms-fronts%3A%3Afront-press-monitor&id=24).

The size of the zip files are slightly different, but downloading each file and running `unzip -l front-press-monitor.zip` on the command line confirms that the unzipped `index.js` files are the same size.

This has also been tested by testing the lambda function in AWS - it ran successfully, as confirmed by these logs:
Screenshot 2023-09-13 at 18 45 55

## Rollout

- [x]  Pause the TeamCity build
- [ ]  Merge this PR
- [ ]  Adjust branch protection rules, requiring the GHA build to pass before merge
- [ ]  Ensure all open PRs are rebased against main to ensure their build runs
- [ ]  Remove the TeamCity build

---

**UPDATE**: I have added the cloudformation.yml template to the files uploaded to riff-raff (previously this would have been done manually) and have configured the riff-raff.yaml file to enable riff-raff to deploy this.
- since making this change I tested by successfully deploying to PROD (there is no CODE stage), verifying that the cloudformation template has been replaced in AWS, and testing the lambda function in AWS, which runs successfully..

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+              uses: actions/checkout@v4
+
+            - name: Install Node
+              uses: actions/setup-node@v3
+              with:
+                  node-version-file: '.nvmrc'
+
+            - name: Install dependencies, build & run tests
+              run: |
+                npm install -g yarn
```
Comment (akash1810) [inline .github/workflows/build.yml:24]: Yarn comes [pre-installed in the environment](https://github.com/actions/runner-images/blob/main/images/linux/Ubuntu2204-Readme.md#package-management), so we should be able to drop this line.

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+                yarn
+                yarn build
+                yarn test
+
+            - name: Configure AWS credentials
+              uses: aws-actions/configure-aws-credentials@v3
```
Comment (akash1810) [inline .github/workflows/build.yml:30]: Let's use the [latest version](https://github.com/aws-actions/configure-aws-credentials/blob/main/CHANGELOG.md) of this action, v4.

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+            - name: Install Node
+              uses: actions/setup-node@v3
+              with:
+                  node-version-file: '.nvmrc'
+
+            - name: Install dependencies, build & run tests
+              run: |
+                npm install -g yarn
+                yarn
```
Comment (akash1810) [inline .github/workflows/build.yml:25]: Let's add the `--frozen-lockfile` flag here to increase determinism.

> If you need reproducible dependencies, which is usually the case with the continuous integration systems, you should pass `--frozen-lockfile` flag.
> – https://classic.yarnpkg.com/lang/en/docs/cli/install/

**Review by akash1810 — COMMENTED** [review]
Looks great! Added a couple of minor comments.

File: .github/workflows/build.yml (hunk 3 above)
Comment (samanthagottlieb) [reply .github/workflows/build.yml]: Thanks for highlighting this, and your comments above 🙏

**Comment by samanthagottlieb** [comment]
UPDATE: I have added the `cloudformation.yml` template to the files uploaded to riff-raff (previously this would have been done manually) and have configured the `riff-raff.yaml` file to enable riff-raff to deploy this.

**Comment by samanthagottlieb** [comment]
> UPDATE: I have added the `cloudformation.yml` template to the files uploaded to riff-raff (previously this would have been done manually) and have configured the `riff-raff.yaml` file to enable riff-raff to deploy this.

Since making this change I tested by successfully deploying to PROD (there is no CODE stage), verifying that the cloudformation template has been replaced in AWS, and testing the lambda function in AWS, which runs successfully.

**Comment by samanthagottlieb** [comment]
As part of rollout I've rebased the open dependabot PRs:
Screenshot 2023-09-22 at 17 27 42
-->
Screenshot 2023-09-22 at 17 33 18

**Pull request merged**